### ISSUE TITLE
ARIA attributes disappearing on BMD2

### ISSUE THREAD

**Issue by AlOneill** [FreeUKGen/FreeBMD2#549]
Unfortunately, **ARIA attributes are disappearing from our pages again**. CEN and REG (both t3) seem to be OK but something has happened on BMD2 t3.

**Comment by AlOneill** [comment]
Unfortunately, **ARIA attributes are disappearing from our pages again**. CEN and REG (both t3) seem to be OK but something has happened on BMD2 t3.

**Comment by DeniseColbert** [comment]
Moved to BMD2 as CEN and REG appear to be OK. 

@Vino-S to look into this

**Comment by Vino-S** [comment]
fix deployed in test3 and production

**Comment by KristinaGadzhieva** [comment]
To check pages that use refinery

**Comment by AlOneill** [comment]
I can **see** the ARIA attributes in place on "Help" but they are **no longer marked up correctly** and so will be of no use to a screen reader user who needs correct markup to understand the structure of the page.

BTW, I was shown an advert for Temu which I understand is a somewhat suspect organisation.

**Comment by AlOneill** [comment]
I also note this problem at the foot of the page on beta:

<img width="1383" alt="Image" src="https://github.com/user-attachments/assets/c23ae6ef-bf09-4b34-a275-7fda4c929a58" />

**Comment by AlOneill** [comment]
And another problem:

<img width="731" alt="Image" src="https://github.com/user-attachments/assets/4499aecd-8d28-4e81-b8d5-b9f253ffa233" />

**Comment by richardofsussex** [comment]
I have attempted to address all of these issues: see beta. (I'm assuming the ARIA error was a labelled-by attribute pointing to a non-existent ID. If there is anything else, please let me know.)

**Comment by richardofsussex** [comment]
The &amp; error was caused by my always writing 'proper' HTML source - ironically the Rails .html_safe command does the escaping for me, so I didn't need to bother! Another point on the learning curve ...

**Comment by AlOneill** [comment]
The ARIA issue is fixed on the Help page. Thank you.

I'm finding it awkward to have to move back up the page to see the "On this page" menu, but perhaps that is just being used to scrolling down a full page. This notion of what constitutes a page is not something I'm comfortable with. Maybe I'm the only person who finds this awkward?

**Comment by AlOneill** [comment]
My other question is why has "On this page" become an h5? This breaks the hierarchy of heading levels.

I'm going to take a look with VoiceOver to see whether this new page structure is accessible to a screen reader.

**Comment by AlOneill** [comment]
It is a while since I used VO, but something has definitely changed. Using Safari, the preferred browser for VO, when I navigate to a new section, such as "Order a certificate", the focus remains in the menu rather than moving to the text in the section. I cannot work out how to get to the text.

@richardofsussex How does Narrator behave?

**Comment by AlOneill** [comment]
On reflection, I think that VO is not being told that the page contents have changed (on navigating to a new section). On selecting "Help", the only headings it recognises are:
- `h1`: Help
- `h5`: On this page
- `h2`: Search

Every other heading (section plus text) is hidden from a screen reader. I believe it will take an `alert` for a SR to realise that the page contents have changed — it is the same issue that makes the form errors listing invisible to a SR user.

**Comment by DeniseColbert** [comment]
Vino to revert the code to the original solution.

**Comment by Vino-S** [comment]
Help page layout is changed. So I believe its not valid any more

**Comment by Vino-S** [comment]
New Help Page layout has been deployed
